api/v2/echo.go

```diff
@@ -168,3 +171,50 @@ func (s *server) Get(c echo.Context) error {
 	c.JSON(http.StatusOK, data)
 	return nil
 }
+
+func (s *server) RWMutexFind(c echo.Context) error {
+	valueStr := c.Param("value")
+	value, err := strconv.Atoi(valueStr)
+	if err != nil {
+		fmt.Println(err)
+		return echo.NewHTTPError(echo.ErrBadRequest.Code, "Invalid value")
+	}
+
+	s.mutex.RLock()
+	defer s.mutex.RUnlock()
+
+	index, ok := s.list.Find(value)
```

Comment:
A suggestion, We can modify this part with:
```
s.mutex.RLock()
index, found := s.list.Find(value)
s.mutex.RUnlock()
```
because this approach could be beneficial as it locks the data only during the find operation
allowing multiple read operations to happen concurrently It ensures thread safety while minimizing the time the lock is held which can improve performance in read heavy scenarios

The problem is that it introduces the potential for errors or bugs if the code between the lock and unlock calls becomes more complex in the future

What do you think? @armineyvazi @alipourhabibi